src/momento/requests/vector_index/item.py

```diff
@@ -19,19 +19,29 @@ class Item:
     vector: list[float]
     """The vector of the item."""
 
-    metadata: dict[str, str] = field(default_factory=dict)
+    metadata: dict[str, str | int | float | bool | list[str]] = field(default_factory=dict)
     """The metadata of the item."""
 
     def to_proto(self) -> pb._Item:
         vector = pb._Vector(elements=self.vector)
         metadata = []
         if self.metadata is not None:
             for k, v in self.metadata.items():
-                if type(v) is not str:
+                if type(v) == str:
```

Comment:
nit: Do you think it will be cleaner if we had a `ser_de_utility` file that handles the serialization/deserialization and this path just call that? Usually when I am reading code in a new repo, it helps seeing both the methods in a single file even if different files (or classes) are accessing them.

Reply:
Another style nit that usually I have seen `switch/case` being used in such cases for readability. But i dont have a strong opinion on that.

Reply:
re serialization utility: maybe? While it's a cross cutting concern and we could centralize it, my first instinct was to place it with the class for cohesion. Can you show me what you mean tomorrow?

re switch/case: this syntax (`match`) wasn't introduced until Python 3.10. Because we have to code to lowest common denominator so that our one library works on older and newer versions, we have to write if-else statements. Unfrotunately.